### ISSUE TITLE
Is Catalyst supported now?

### ISSUE THREAD

**Issue by dev-two** [realm/realm-js#2883]
This issue was mentioned before – #2629 – but answer was never described properly.

## Goals

Enable Mac as target for the project.

## Expected Results

App builds for Mac

## Actual Results

Build fails with
```
[...]node_modules/realm/vendor/realm-ios/librealm-ios.a(placeholder.o), building for Mac Catalyst, but linking in object file built for iOS Simulator, file '[...]node_modules/realm/vendor/realm-ios/librealm-ios.a' for architecture x86_64
```

## Version of Realm and Tooling

- Realm JS SDK Version: 4.0.0-beta.1
- Node or React Native: 0.61.5
- Client OS & Version: macOS 10.15.4, Xcode 11.4.1
- Which debugger for React Native: None

**Comment by juhasuni** [comment]
I haven't tested with Catalyst, but you should check this issue #2907. It seems like Realm Core libraries are already available for iOS, tvOS, MacOS and Catalyst, so it might be possible to add the support with very few modifications. 

When CocoaPods installs RealmJS, it downloads some pre-built libraries and those libraries contain versions for various platforms. After installing this module using CocoaPods, you should check the contents of `node_modules/realm/vendor/realm-ios`. You'll see, that there are `librealm-ios.a` and `librealm-maccatalyst.a`. I didn't have time to test Catalyst, as we're more interested in MacOS support via the new `react-native-macos` developed by Microsoft.

**Comment by kneth** [comment]
@dev-two Catalyst is not supported, and we haven't a plan for it yet.  As @juhasuni mentions, Realm Core is built for it (for Realm Cocoa), and we will appreciate if you can describe what it will require for us to support it.

**Comment by marcosrdz** [comment]
I changed their Podspec to this:

  s.ios.vendored_libraries = 'vendor/realm-ios/librealm-maccatalyst.a', 'vendor/realm-ios/librealm-parser-maccatalyst.a'

**Comment by marcosrdz** [comment]
the xcframework used on 10.3.0 or newer have these folders:

ios-arm64_armv7
ios-arm64_i386_x86_64-simulator

However, Realm used to support Mac Catalyst up until 6.0.0.

An example folder of a xcframework with support for Catalyst would include a folder like this:

ios-x86_64-maccatalyst/

6.0.0 no longer works in the new big sur release, making any Catalyst app using this library crash on runtime.

**Comment by marcosrdz** [comment]
Support added in 10.6.0. Confirmed.

**Comment by takameyer** [comment]
This has been released in version 10.6.0.  I will now close this issue.
